More clarity on efficiency and idle. Include mathematical definitions. (#553)
kubecost/docs: efficiency-idle.md

![The Allocations view aggregated by namespace, which shows efficiency & idle](images/efficiency-idle.png)

## Efficiency

Pod resource efficiency is defined as the resource utilization versus the resource request over a given time window. It is cost-weighted and can be expressed as follows:

> *(((CPU Usage / CPU Requested) * CPU Cost) + ((RAM Usage / RAM Requested) * RAM Cost))) / (RAM Cost + CPU Cost)*
>
> where \
> *CPU Usage = rate(container\_cpu\_usage\_seconds\_total) over the time window* \
> *RAM Usage = avg(container\_memory\_working\_set\_bytes) over the time window*

For example, if a pod is requesting 2CPU and 1GB, using 500mCPU and 500MB, CPU on the node costs $10/CPU, and RAM on the node costs $1/GB, we have ((0.5/2) \* 20 + (0.5/1) \* 1) / (20 + 1) = 5.5 / 21 = 26%

## Idle

Cluster idle cost is defined as the difference between the cost of allocated resources and the cost of the hardware they run on. Allocation is defined as the max of usage and requests. It can also be expressed as follows:

> *idle_cost = sum(node_cost) - (cpu_allocation_cost + ram_allocation_cost + gpu_allocation_cost)*
>
> where \
> *allocation = max(request, usage)*

> So, idle costs can also be thought of as the cost of the space that the Kubernetes scheduler could add pods without disrupting any workloads in but is not currently.

Reviewer: This is confusing to me. How about if we reword it to something like this:
```suggestion
So, idle costs can also be thought of as the cost of the space that the Kubernetes scheduler could schedule pods, without disrupting any existing workloads, but it is not currently.
```

Author: Good suggestion, merged!
